### DISCUSSION TITLE
Can Microcks read Excel data or data from db to populate response

### DISCUSSION THREAD

**Discussion by Athira31** [microcks/microcks#987] [Q&A] [answered]
For creating dynamic responses in microcks can we read and map values from external data source such as db,Excel,dile etc

**Comment by lbroudoux** [comment, marked as answer]
I recommend looking at the [`SCRIPT` dispatcher](https://microcks.io/documentation/using/advanced/dispatching/#script-dispatcher) feature for creating dynamic responses. We don't directly provide connectors for DB or file, but using the Groovy languages facilities you may certainly find a way to interact with your datasource. We also provide a way to extend Microcks with your own Groovy or Java lib for that purpose. See https://microcks.io/documentation/administrating/externals/

Once you have fetched data from the sources, you may use [Context expression](https://microcks.io/documentation/using/advanced/templates/#context-expression) to integrate this data into your response template.

Hope it helps!

**Reply by lbroudoux** [reply]
Hey @Athira31! Did you have the opportunity to check ^^?
